spring-cloud-dataflow-audit/src/main/java/org/springframework/cloud/dataflow/audit/service/DefaultAuditRecordService.java

```diff
@@ -38,74 +41,101 @@
-	}
+    private static final Logger logger = LoggerFactory.getLogger(DefaultAuditRecordService.class);
+
+    private final AuditRecordRepository auditRecordRepository;
+
+    private final ObjectMapper objectMapper;
+    private final ArgumentSanitizer sanitizer;
+
+    public DefaultAuditRecordService(AuditRecordRepository auditRecordRepository) {
+        this(auditRecordRepository, null);
+    }
+
+    public DefaultAuditRecordService(AuditRecordRepository auditRecordRepository, ObjectMapper objectMapper) {
+        Assert.notNull(auditRecordRepository, "auditRecordRepository must not be null.");
+        this.auditRecordRepository = auditRecordRepository;
+        if (objectMapper == null) {
+            objectMapper = new ObjectMapper();
+            objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        }
+        this.objectMapper = objectMapper;
+        this.sanitizer = new ArgumentSanitizer();
+    }
+
+    @Override
+    public AuditRecord populateAndSaveAuditRecord(AuditOperationType auditOperationType,
+                                                  AuditActionType auditActionType,
+                                                  String correlationId, String data, String platformName) {
+        Assert.notNull(auditActionType, "auditActionType must not be null.");
+        Assert.notNull(auditOperationType, "auditOperationType must not be null.");
+
+        final AuditRecord auditRecord = new AuditRecord();
+        auditRecord.setAuditAction(auditActionType);
+        auditRecord.setAuditOperation(auditOperationType);
+        auditRecord.setCorrelationId(correlationId);
+        auditRecord.setAuditData(data);
+        auditRecord.setPlatformName(platformName);
+        return this.auditRecordRepository.save(auditRecord);
+    }
+
+    @Override
+    public AuditRecord populateAndSaveAuditRecordUsingMapData(
+            AuditOperationType auditOperationType,
+            AuditActionType auditActionType,
+            String correlationId, Map<String, Object> data,
+            String platformName
+    ) {
+        String dataAsString;
+        try {
+            Map<String, Object> sanitizedData = sanitizeMap(data);
+            dataAsString = objectMapper.writeValueAsString(sanitizedData);
+        } catch (JsonProcessingException e) {
+            logger.error("Error serializing audit record data.  Data = " + data);
+            dataAsString = "Error serializing audit record data.  Data = " + data;
+        }
+        return this.populateAndSaveAuditRecord(auditOperationType, auditActionType, correlationId, dataAsString, platformName);
+    }
+
+    private Map<String, Object> sanitizeMap(Map<String, Object> data) {
+        final Map<String, Object> result = new HashMap<>();
+        data.entrySet()
+                .stream()
+                .map(this::sanitize)
+                .forEach(entry -> result.put(entry.getKey(),entry.getValue()));
+        return result;
```

Comment:
Now that we have to use `forEach`, I would instead use the one directly on the map and then make `sanitize` just take a k/v. It was only using `Map.Entry` to take advantage of the method reference in the `map` on the stream really. This really simplifies it down to this now:
```java
Map<String, Object> result = new HashMap<>();
data.forEach((k, v) -> result.put(k, sanitize(k, v)));
return result;
```